tests/measurements/test_classical_shadow.py

```diff
@@ -223,6 +223,159 @@ def test_expval_same_rng(self):
         assert res1 == res2
 
 
+class TestProcessDensityMatrix:
+    """Unit tests for process_state_with_shots for the classical_shadow
+    and shadow_expval measurements"""
+
+    def test_shape_and_dtype(self):
+        """Test that the shape and dtype of the measurement is correct"""
+        mp = qml.classical_shadow(wires=[0, 1])
+        state = np.ones((2, 2)) / 2
+        dm = np.outer(state, state).reshape((2,) * 4)
+        res = mp.process_density_matrix_with_shots(dm, qml.wires.Wires([0, 1]), shots=100)
+
+        assert res.shape == (2, 100, 2)
+        assert res.dtype == np.int8
+
+        # test that the bits are either 0 and 1
+        assert np.all(np.logical_or(res[0] == 0, res[0] == 1))
+
+        # test that the recipes are either 0, 1, or 2 (X, Y, or Z)
+        assert np.all(np.logical_or(np.logical_or(res[1] == 0, res[1] == 1), res[1] == 2))
+
+    def test_wire_order(self):
+        """Test that the wire order is respected"""
+        state = np.array([[1, 1], [0, 0]]) / np.sqrt(2)
+        dm = np.outer(state, state).reshape((2,) * 4)
+
+        mp = qml.classical_shadow(wires=[0, 1])
+        res = mp.process_density_matrix_with_shots(dm, qml.wires.Wires([0, 1]), shots=1000)
+
+        assert res.shape == (2, 1000, 2)
+        assert res.dtype == np.int8
+
+        # test that the first qubit samples are all 0s when the recipe is Z
+        assert np.all(res[0][res[1, ..., 0] == 2][:, 0] == 0)
+
+        # test that the second qubit samples contain 1s when the recipe is Z
+        assert np.any(res[0][res[1, ..., 1] == 2][:, 1] == 1)
```

Comment:
Should we fix seeds if we rely on this random sampling yielding a 1?

Reply:
seeded at https://github.com/PennyLaneAI/pennylane/pull/6748/commits/29f32747b92727108f267fb47a39f12b3186ebdd and https://github.com/PennyLaneAI/pennylane/pull/6748/commits/6c7a834d6dea7675507e2d825a18e7b01ca184d9

Reply:
Thanks, amazing :)